R/format_badges.R

```diff
@@ -0,0 +1,79 @@
+format_badge_url <- function(display, path) {
+  glue::glue("[{display}]({path})")
+}
+
+format_name_badge <- function(repo_path) {
+  repo_org <- strsplit(repo_path, "/", fixed = TRUE)[[1L]][[1L]]
+  repo_name <- strsplit(repo_path, "/", fixed = TRUE)[[1L]][[2L]]
+
+  format_badge_url(
+    display = repo_name,
+    path = glue::glue("https://{tolower(repo_org)}.github.io/{repo_name}/")
+  )
+}
+
+format_lifecycle_badge <- function(lifecycle) {
+  stopifnot(
+    lifecycle %in% c("experimental", "stable", "deprecated", "superseded")
+  )
+
+  lifecycle_badge_url <- switch(
+    lifecycle,
+    experimental = "https://lifecycle.r-lib.org/reference/figures/lifecycle-experimental.svg", # nolint: line_length_linter
+    stable = "https://lifecycle.r-lib.org/reference/figures/lifecycle-stable.svg", # nolint: line_length_linter
+    deprecated = "https://lifecycle.r-lib.org/reference/figures/lifecycle-deprecated.svg", # nolint: line_length_linter
+    superseded = "https://lifecycle.r-lib.org/reference/figures/lifecycle-superseded.svg" # nolint: line_length_linter
+  )
+
+  format_badge_url(
+    display = glue::glue(
+      "![]({lifecycle_badge_url})"
+    ),
+    path = glue::glue(
+      "https://lifecycle.r-lib.org/articles/stages.html#{lifecycle}"
+    )
+  )
+}
+
+format_status_badge <- function(repo_path, ci_check) {
+  format_badge_url(
+    display = glue::glue(
+      "![](https://github.com/{repo_path}/actions/workflows/{ci_check}/badge.svg?branch=main)" # nolint: line_length_linter
+    ),
+    path = glue::glue(
+      "https://github.com/{repo_path}/actions/workflows/{ci_check}?query=branch%3Amain" # nolint: line_length_linter
```

Comment:
I think all of the relevant repos use `main` not `master`, so this is not a problem?